docs/gradle-plugin.md

```diff
@@ -0,0 +1,270 @@
+
+The follow inputs will be picked by the `precompileJte` task:
+
+| Parameter               | Description                                                                       | Default                                                        |
+|-------------------------|-----------------------------------------------------------------------------------|----------------------------------------------------------------|
+| `sourceDirectory`       | The directory where template files are located                                    | `src/main/jte`                                                 |
+| `targetDirectory`       | The directory where compiled classes should be written to                         | `jte-classes`                                                  |
+| `compilePath`           | The compile-classpath to use                                                      | `sourceSets.main.runtimeClasspath`                             |
+| `contentType`           | The content type of all templates. Either `Plain` or `Html`                       | `Html`                                                         |
+| `trimControlStructures` | Trims control structures, resulting in prettier output                            | `false`                                                        |
+| `htmlTags`              | Intercepts the given html tags during template compilation                        | None                                                           |
+| `htmlPolicyClass`       | An [`HtmlPolicy`][html-policy] used to check the parsed HTML at compile time      | None                                                           |
+| `htmlCommentsPreserved` | If HTML comments should be preserved in the output                                | `false`                                                        |
+| `binaryStaticContent`   | If to generate a [binary content](binary-rendering.md) resource for each template | `false`                                                        |
+| `compileArgs`           | Sets additional compiler arguments for `jte` templates.                           | None                                                           |
+| `kotlinCompileArgs`     | Sets additional compiler arguments for `kte` templates                            | None                                                           |
+| `packageName`           | The package name, where template classes are generated to                         | [`Constants.PACKAGE_NAME_PRECOMPILED`][constants-package-name] |
+
+!!! info "About `htmlPolicyClass`"
+
+    The `htmlPolicyClass` will default to `gg.jte.html.OwaspHtmlPolicy` if the content type is `Html`.
+
+!!! warning "Clean all before precompiling"
+
+    The `precompileJte` task cleans the directory containing the compiled template classes every time it runs. In your local development environment, it may make more sense to use [hot reloading](hot-reloading.md).
```

Comment:
Same comment here as in the other PR: The hint to hot reloading seems unrelated to the `Clean all before precompiling` warning.